File: src/constants/urls.ts
/**
 * Centralized URL Configuration for Frontend
 * All external URLs and API endpoints should be defined here
 */

// GitHub Repository URLs
export const GITHUB_URLS = {
    REPO_OWNER: 'akash2061',
    REPO_NAME: 'NoteBook-App',
    RELEASES: 'https://github.com/akash2061/NoteBook-App/releases',
    PROFILE: 'https://github.com/akash2061',
    CODE_CANVAS: 'https://github.com/akash2061/Code_Canvas',
    COSMOSECURE_ORG: 'https://github.com/CosmoSecure',
} as const;

// Social Media URLs
export const SOCIAL_URLS = {
    LINKEDIN: 'https://www.linkedin.com/in/akash2061/',
    TWITTER: '',
    YOUTUBE: '',
} as const;

// Project URLs
export const WEBSITE_URLS = {
    WEBSITE: 'https://cosmosecure.vercel.app',
    ABOUT: 'https://cosmosecure.vercel.app/about',
    DOCS: 'https://cosmosecure.vercel.app/docs',
    SIGNUP: 'https://cosmosecure.vercel.app/signup',
    FORGOT_PASSWORD: 'https://cosmosecure.vercel.app/forgot-password',
} as const;

// API Endpoints
export const API_URLS = {
    // Email breach check API
    EMAIL_BREACH: 'https://api.xposedornot.com/v1/breach-analytics',
} as const;

// Documentation URLs
export const DOCS_URLS = {
    NODEJS: 'https://nodejs.org/',
    RUST: 'https://www.rust-lang.org/tools/install',
    TAURI: 'https://v2.tauri.app/',
    CARGO_DOCS: 'https://doc.rust-lang.org/cargo/reference/manifest.html',
} as const;

// Development URLs
export const DEV_URLS = {
    LOCALHOST: 'http://localhost:1420',
    VITE_HMR_HOST: 'http://localhost:1421',
} as const;

// Helper function to build GitHub API URL
export const buildGitHubApiUrl = (endpoint: string) => {
    return `https://api.github.com/repos/${GITHUB_URLS.REPO_OWNER}/${GITHUB_URLS.REPO_NAME}${endpoint}`;
};

// Helper function to build email breach URL with email
export const buildEmailBreachUrl = (email: string) => {
    return `${API_URLS.EMAIL_BREACH}?email=${encodeURIComponent(email)}`;
};

export default {
    GITHUB_URLS,
    SOCIAL_URLS,
    WEBSITE_URLS,
    API_URLS,
    DOCS_URLS,
    DEV_URLS,
    buildGitHubApiUrl,
    buildEmailBreachUrl,
};

File: src-tauri/src/urls.rs
/// Centralized URL Configuration for Rust Backend (src-tauri)
/// All external URLs and API endpoints should be defined here

/// GitHub Repository Configuration
pub mod github {
    pub const REPO_OWNER: &str = "akash2061";
    pub const REPO_NAME: &str = "NoteBook-App";
    pub const RELEASES_API: &str = "https://api.github.com/repos/akash2061/NoteBook-App/releases";

    /// Build GitHub API URL for a specific endpoint
    pub fn build_api_url(endpoint: &str) -> String {
        format!(
            "https://api.github.com/repos/{}/{}{}",
            REPO_OWNER, REPO_NAME, endpoint
        )
    }
}

/// External API Endpoints
pub mod api {
    /// Email breach check API base URL
    pub const EMAIL_BREACH_BASE: &str = "https://api.xposedornot.com/v1/breach-analytics";

    /// Build email breach URL with email parameter
    pub fn build_email_breach_url(email: &str) -> String {
        format!("{}?email={}", EMAIL_BREACH_BASE, urlencoding::encode(email))
    }
}

/// User Agent Strings
pub mod user_agent {
    pub const COSMOSECURE_APP: &str = "CosmoSecure-App";
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_build_github_api_url() {
        let url = github::build_api_url("/releases");
        assert_eq!(
            url,
            "https://api.github.com/repos/akash2061/NoteBook-App/releases"
        );
    }

    #[test]
    fn test_build_email_breach_url() {
        let url = api::build_email_breach_url("test@example.com");
        assert!(url.contains("api.xposedornot.com"));
        assert!(url.contains("test%40example.com"));
    }
}
